lib/core/form/components/widgets/text/text.widget.spec.ts

```diff
@@ -75,14 +81,124 @@ describe('TextWidgetComponent', () => {
                 });
             }));
 
+            it('should be able to set label property', () => {
+                widget.field = new FormFieldModel(new FormModel({ taskId: 'fake-task-id' }), {
+                    id: 'text-id',
+                    name: 'text-name',
+                    value: '',
+                    type: FormFieldTypes.TEXT,
+                    readOnly: false
+                });
+
+                fixture.detectChanges();
+                const textWidgetLabel = element.querySelector('label');
+                expect(textWidgetLabel.innerText).toBe('text-name');
+            });
+
+            it('should be able to set a Text Widget as required', async () => {
+                widget.field = new FormFieldModel(new FormModel({ taskId: 'fake-task-id' }), {
+                    id: 'text-id',
+                    name: 'text-name',
+                    value: '',
+                    type: FormFieldTypes.TEXT,
+                    readOnly: false,
+                    required: true
+                });
+
+                fixture.detectChanges();
```

Comment:
if you don't put await here the test won't wait for the detectChanges to be executed.

Reply:
is it possible you're thinking of this as the whenStable method?

Reply:
no, detectChanges returns a promise if you don't wait it with await or waiting the rendering with whenStable this won't render the page correctly.